src/widgets/MultiSelectList.js

```diff
@@ -0,0 +1,235 @@
+      );
+    }
+
+    return data;
+  };
+
+  /**
+   * Filters an array by two filter properties, and user input query text.
+   * Ignores case. Querying a realm result with filtered is more performant,
+   * so have two cases for each.
+   */
+  filterArrayData = options => {
+    const { primaryFilterProperty, secondaryFilterProperty } = this.props;
+    const { queryText } = this.state;
+
+    const regexFilter = RegExp(queryText, 'i');
+
+    return options.filter(
+      optionItem =>
+        regexFilter.test(optionItem[primaryFilterProperty]) ||
+        regexFilter.test(optionItem[secondaryFilterProperty])
+    );
+  };
+
+  /**
+   * Delegator of filtering process. Check if the object is a realm
+   * object (has the filtered member) or if it is an array. Otherwise,
+   * return an empty list to display.
+   */
+  getData = () => {
+    const { options, primaryFilterProperty, queryString } = this.props;
+    if (options && options.filtered && queryString) return this.filterResultData(options);
+    if (Array.isArray(options) && primaryFilterProperty) return this.filterArrayData(options);
+    return [];
+  };
+
+  /**
+   * Render single item of the FlatList
+   *
+   * @prop  {object}     data   Single item data.
+   *
+   * @returns {component} ResultRow
+   *
+   */
+  renderItem = data => {
+    const { renderLeftText, renderRightText, showCheckIcon } = this.props;
+    return (
+      <ResultRow
+        data={data}
+        onPress={() => this.onSelect(data)}
+        isSelected={() => this.isSelected(data)}
+        renderLeftText={renderLeftText}
+        renderRightText={renderRightText}
+        showCheckIcon={showCheckIcon}
+      />
+    );
+  };
+
+  /**
+   * On select FlatList item update selected state
+   *
+   * @prop  {object}     data   Single item data.
+   */
+  onSelect = data =>
+    this.setState(prevState =>
+      prevState.selected.indexOf(data.item.id) === -1
+        ? { selected: [...prevState.selected, data.item.id] }
+        : { selected: prevState.selected.filter(i => i !== data.item.id) }
+    );
+
+  /**
+   * Check if the single item is selected or not
+   *
+   * @prop  {object}     data   Single item data.
+   *
+   * @returns {bool}    selected state of the item
+   */
+  isSelected = data => {
+    const { selected } = this.state;
+    return !(selected.indexOf(data.item.id) === -1);
+  };
+
+  render() {
+    const { placeholderText, onConfirmSelections } = this.props;
+    const { selected } = this.state;
+    const data = this.getData();
+
+    return (
+      <View style={localStyles.container}>
+        <SearchBar
+          autoCapitalize="none"
+          autoCorrect={false}
+          autoFocus
+          color="white"
+          onChange={text => this.setState({ queryText: text })}
+          placeholder={placeholderText}
+          placeholderTextColor="white"
+          style={[localStyles.text, localStyles.searchBar]}
+        />
+        {data.length > 0 && (
```

Comment:
this causing this situation is not great

![image](https://user-images.githubusercontent.com/35858975/69175310-5f347880-0b68-11ea-8649-10489fbb7103.png)


options: Disable the button, show some "none available" warning ? Any others?

(New issue)

Reply:
@joshxg We could disable/hide the `Add from masterlist` button itself if the customer/thisStore does not have Masterlists.

Reply:
@sworup I think its the best option.. just then the user doesnt know what is going on.. why is it disabled or invisible

Reply:
you can use `AndroidToast` I think, when there is none, to say "No masterlists sorry bro!"

Reply:
Disable might confuse users yes, but if remove it from the interface the user would not bother, I think. That's how Steve Jobs did it. 😄 

We can disable the button and change label to No masterlist exists or something.

Reply:
Conversation on Telegram between @joshxg @raviSussol and Ujwal about this. Maybe we move this to a new issue.

![2019-11-20 12 26 32](https://user-images.githubusercontent.com/683452/69215166-05947400-0b91-11ea-9026-d846f2f06987.jpg)

Reply:
I'm fine if it's a new issue - but I don't think we will be waiting for this TMG (I dunno what that is? Ivory coast?) - If we get instructions that there is some change request,  then we can do changes, but I have not been told anything?

And I don't know anything about any 'proposed changes' - who proposed the changes? What were the changes?

none of this makes sense to me .. sorry ..

Reply:
I think making the button invisible in this case is bad. A toast message is a good mechanism for user feedback in this case. To get this PR merged, that addition should be its own issue. 

We generally want users to be using the button where possible, so it's good to know it's there. If users get some feedback that master lists aren't setup (without feedback they may conclude it's broken), they know it can be done and request it from admins if appropriate. Having it hidden for some but not all customers with no explanation doesn't achieve that.

In the case of dispensing to patients, we're implementing that as it's own distinct functionality over the next couple months anyway.

Reply:
@joshxg @Chris-Petty I showed the message below for empty masterlist list. I think that should cover it for now. We can do disable/hide button or something else in another issue.

![Screenshot 2019-11-21 at 15 31 16](https://user-images.githubusercontent.com/683452/69326513-03efac80-0c74-11ea-83c2-b2ac05b99a0e.jpg)

Reply:
Resolved for this issue with 06caa51a